src/proxy_lib/proxy_lib.c

```diff
@@ -107,34 +132,97 @@ static __TLS int was_called_from_umfPool = 0;
 /*** The constructor and destructor of the proxy library *********************/
 /*****************************************************************************/
 
+#ifndef _WIN32
+static size_t get_size_threshold(void) {
+    char *str_threshold = utils_env_var_get_str("UMF_PROXY", "size.threshold=");
+    if (!str_threshold) {
+        return 0;
+    }
+
+    // move to the beginning of the number
+    str_threshold += strlen("size.threshold=");
+    // find ';' at the end
+    char *end = strstr(str_threshold, ";");
+    if (end) {
+        // replace ';' with '\0' to mark end of the string
+        *end = '\0';
+    }
+
```

Comment:
we could have a test for checking this parsing

Reply:
I would have to move this function to utils, it cannot be tested here as a part of proxy lib.

Reply:
> we could have a script or part of CI, that will run multiple configurations, like:
> 
> ```
> UMF_PROXY="size.threshold=XYZ;<other var>" LD_PRELOAD=./lib/libumf_proxy.so ctest --output-on-failure
> UMF_PROXY="<other var>;size.threshold=ABC" LD_PRELOAD=./lib/libumf_proxy.so ctest --output-on-failure
> etc...
> ```

I am afraid it would take too much time ...

Reply:
Added tests for this specific function: https://github.com/oneapi-src/unified-memory-framework/pull/883/commits/3c04a4f9fecbe7897f6df0b430b6502fafec0163
Done